lib/commands/nexus.rb

```diff
@@ -110,15 +110,22 @@ def send_gem
       request.add_field("Authorization", authorization.strip) if authorization
     end
 
+    exit_code = 0
     case response.code
-    when "401"
-      say "Unauthorized"
     when "400"
       say "something went wrong - maybe (re)deployment is not allowed"
+    when "401"
+      say "Unauthorized"
     when "500"
       say "something went wrong"
     else
       say response.message
     end
+
+    if !response.kind_of? Net::HTTPSuccess
+      exit_code = 1
+      exit exit_code
+    end
+    exit_code
```

Comment:
Not a bad change.  Since the behavior was already exit with a "success" status on failure then the only code that needs to be added is:

```ruby
+    if !response.kind_of? Net::HTTPSuccess
+      exit 1
+    end
```